docs/en/upload.md

```diff
@@ -56,21 +56,63 @@ function startup()
     end
 end
 
+-- Define WiFi station event callbacks 
+wifi_connect_event = function(T) 
+  print("Connection to AP("..T.SSID..") established!")
+  print("Waiting for ip...") 
+end
+
+wifi_got_ip_event = function(T) 
+  print("Wifi connection is ready! IP address is: "..T.IP)
+  print("Startup will resume momentarily, you have 3 seconds to abort.")
+  print("Waiting...") 
+  tmr.create():alarm(3000, tmr.ALARM_SINGLE, startup)
+end
+
+wifi_disconnect_event = function(T)
+  if T.reason == wifi.eventmon.reason.ASSOC_LEAVE then 
+    --the station has disassociated from a previously connected AP
+    return 
+  end
+  -- total_tries: how many times the station will attempt to connect to the AP.
+  local total_tries = 3
```

Comment:
My Comcast xfinity gateway(Arris technicolor tc8305c) takes about 2.5 min to reboot and the interval between `NO_AP_FOUND` disconnect events is about 2.25 seconds, so I think a value of 75 should cover it.